### PR TITLE
azurerm_postgresql_server - Fix legacy database ids handling

### LINKED ISSUE #17763: azurerm 3.15.1: unable to parse existing `azurerm_postgresql_database` resource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.15.1

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_database

### Terraform Configuration Files

```hcl
resource "azurerm_postgresql_database" "this" {
  for_each            = local.filtered_pg_instances
  name                = lookup(local.filtered_pg_instances[each.key], "database_name", each.key)
  resource_group_name = azurerm_postgresql_server.this[each.key].resource_group_name
  server_name         = azurerm_postgresql_server.this[each.key].name
  charset             = "UTF8"
  collation           = "English_United States.1252"
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: parsing Postgres Server ID : parsing "/subscriptions/REDACTED/resourceGroups/smc-prod-australiasoutheast/providers/Microsoft.DBForPostgreSQL/servers/REDACTED": parsing segment "staticMicrosoftDBforPostgreSQL": expected the segment "Microsoft.DBForPostgreSQL" to be "Microsoft.DBforPostgreSQL"
│ 
│   with module.stack.azurerm_postgresql_server.this["REDACTED"],
│   on ../../modules/seismic-az-stack/postgresql.tf line 135, in resource "azurerm_postgresql_server" "this":
│  135: resource "azurerm_postgresql_server" "this" {
│ 
╵
```

### Expected Behaviour

```
$ terraform plan
No changes. Your infrastructure matches the configuration.
```

### Actual Behaviour

Getting an error when doing `terraform plan`:

```
╷
│ Error: parsing Postgres Server ID : parsing "/subscriptions/REDACTED/resourceGroups/smc-prod-australiasoutheast/providers/Microsoft.DBForPostgreSQL/servers/REDACTED": parsing segment "staticMicrosoftDBforPostgreSQL": expected the segment "Microsoft.DBForPostgreSQL" to be "Microsoft.DBforPostgreSQL"
│ 
│   with module.stack.azurerm_postgresql_server.this["REDACTED"],
│   on ../../modules/seismic-az-stack/postgresql.tf line 135, in resource "azurerm_postgresql_server" "this":
│  135: resource "azurerm_postgresql_server" "this" {
│ 
╵
```

### Steps to Reproduce

1. Set the azurerm provider to v3.15.1
1. terraform plan

### Important Factoids

_No response_

### References

This is the same issue that was solved by 3.15.1 in https://github.com/hashicorp/terraform-provider-azurerm/issues/17722 but for a different resource.

### PR DESCRIPTION
**Pull request by antonblr**
Fixes #17763

This fix is similar to https://github.com/hashicorp/terraform-provider-azurerm/pull/17755, but for `databases`.

Tested by running `terraform plan` on legacy (version 11) Azure Postgres resource (the same example as in #17763):

```hcl
resource "azurerm_postgresql_database" "this" {
  for_each            = local.filtered_pg_instances
  name                = lookup(local.filtered_pg_instances[each.key], "database_name", each.key)
  resource_group_name = azurerm_postgresql_server.this[each.key].resource_group_name
  server_name         = azurerm_postgresql_server.this[each.key].name
  charset             = "UTF8"
  collation           = "English_United States.1252"
}
```
* `terraform plan` (no errors)
```hcl
Terraform will perform the following actions:

  # module.stack.azurerm_postgresql_server.this["REDACTED"] will be updated in-place
  ~ resource "azurerm_postgresql_server" "this" {
        id                                = "/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.DBforPostgreSQL/servers/REDACTED"
        name                              = "REDACTED"
      ~ sku_name                          = "GP_Gen5_8" -> "GP_Gen5_2"

        # (14 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### CODE REVIEW

File: internal/services/postgres/postgresql_database_resource.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"log"
 	"time"
 
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres/migration"
+
```
Comment (catriona-m) [inline internal/services/postgres/postgresql_database_resource.go:9]: looks like there's an extra space here

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @antonblr - just left one minor comment, but can have another look at this once that is addressed. Thanks!

File: internal/services/postgres/postgresql_database_resource.go (hunk 1 above)
Comment (antonblr) [reply internal/services/postgres/postgresql_database_resource.go]: Thanks @catriona-m. I thought there was some convention here - https://github.com/hashicorp/terraform-provider-azurerm/blob/620839508fa6f8a0e46c6129b94b64a42aa00b55/internal/services/postgres/postgresql_server_resource.go#L10-L13

Fixed in 0c87b1c (will squash afterwards). Is there a linter for that, btw?

Comment (katbyte) [reply internal/services/postgres/postgresql_database_resource.go]: there is: `make goimports` is what we should go by

**Review by katbyte — APPROVED** [review]
Thanks @antonblr - LGTM 🏗️

**Pull request merged**